l1-contracts/src/core/Decoder.sol

```diff
@@ -15,55 +15,58 @@ pragma solidity >=0.8.18;
+ *  | 0xb8                          | 0x20       | startPublicDataTreeRoot
+ *  | 0xd8                          | 0x20       | startL1ToL2MessagesTreeSnapshot.root
+ *  | 0xf8                          | 0x04       | startL1ToL2MessagesTreeSnapshot.nextAvailableLeafIndex
+ *  | 0xfc                          | 0x20       | startTreeOfHistoricL1ToL2MessagesTreeRootsSnapshot.root
+ *  | 0x11c                         | 0x04       | startTreeOfHistoricL1ToL2MessagesTreeRootsSnapshot.nextAvailableLeafIndex
+ *  | 0x120                         | 0x20       | endPrivateDataTreeSnapshot.root
+ *  | 0x140                         | 0x04       | endPrivateDataTreeSnapshot.nextAvailableLeafIndex
+ *  | 0x144                         | 0x20       | endNullifierTreeSnapshot.root
+ *  | 0x164                         | 0x04       | endNullifierTreeSnapshot.nextAvailableLeafIndex
+ *  | 0x168                         | 0x20       | endContractTreeSnapshot.root
+ *  | 0x188                         | 0x04       | endContractTreeSnapshot.nextAvailableLeafIndex
+ *  | 0x18c                         | 0x20       | endTreeOfHistoricPrivateDataTreeRootsSnapshot.root
+ *  | 0x1ac                         | 0x04       | endTreeOfHistoricPrivateDataTreeRootsSnapshot.nextAvailableLeafIndex
+ *  | 0x1b0                         | 0x20       | endTreeOfHistoricContractTreeRootsSnapshot.root
+ *  | 0x1d0                         | 0x04       | endTreeOfHistoricContractTreeRootsSnapshot.nextAvailableLeafIndex
+ *  | 0x1d4                         | 0x20       | endPublicDataTreeRoot
+ *  | 0x1f4                         | 0x20       | endL1ToL2MessagesTreeSnapshot.root
+ *  | 0x214                         | 0x04       | endL1ToL2MessagesTreeSnapshot.nextAvailableLeafIndex
+ *  | 0x218                         | 0x20       | endTreeOfHistoricL1ToL2MessagesTreeRootsSnapshot.root
+ *  | 0x238                         | 0x04       | endTreeOfHistoricL1ToL2MessagesTreeRootsSnapshot.nextAvailableLeafIndex
+ *  | 0x23c                         | 0x04       | len(newCommitments) denoted x
+ *  | 0x240                         | x          | newCommits
+ *  | 0x240 + x                     | 0x04       | len(newNullifiers) denoted y
+ *  | 0x244 + x                     | y          | newNullifiers
+ *  | 0x244 + a + b                 | 0x04       | len(newPublicDataWrites) denoted c
+ *  | 0x248 + a + b                 | c          | newPublicDataWrites (each element 64 bytes)
+ *  | 0x248 + a + b + c             | 0x04       | len(newL2ToL1msgs) denoted d
+ *  | 0x24c + a + b + c             | d          | newL2ToL1msgs (each element 32 bytes)
+ *  | 0x24c + a + b + c + d         | 0x04       | len(newContracts) denoted e
+ *  | 0x250 + a + b + c + d         | e          | newContracts (each element 32 bytes)
+ *  | 0x250 + a + b + c + d + e     | e          | newContractData (each element 52 bytes)
+ *  | 0x250 + a + b + c + d + e     | 0x04       | len(l1ToL2Messages) denoted f
```

Comment:
should be 2e no? Since we are adding e twice

Reply:
The notation right now is somewhat sloppy as neither a,b,c,d,e,f are byte lengths. Might just fix that entirely and make the width of the table grow.